### ISSUE TITLE
Steam changed app ticket format, current module does not work with new format

### ISSUE THREAD

**Issue by SkywallStudiosLLC** [DoctorMcKay/node-steam-appticket#5]
![image](https://github.com/user-attachments/assets/aae189d4-6909-4622-953e-9eb0e4ee15a2)

Hi, I recently encountered an issue with this module related to Steam changing their encrypted app ticket format. When checking if the decrypted app ticket is null it is returning true. I made a support ticket with Steam, they mentioned that they recently updated their app ticket format. Luckily for me they were able to revert the change for my specific app so it is working again. However this means that this module will need to be updated to work with the new ticket format.

I know it's been a while since this was last updated but I hope it can be.

**Comment by DoctorMcKay** [comment]
Fixed in 2.0.0-beta.1 and backported in 1.0.2.

2.0.0-beta.1 should be stable; if you could give it a try and let me know how it works for you that would be great, then I can promote it to general release.

**Comment by ckap** [comment]
Can't speak for Skywall, but this resolved the issues we were having on our backend. Thank you so much!

**Comment by DoctorMcKay** [comment]
Thanks, it's published under latest as 2.0.0 now.

**Comment by tsa96** [comment]
Yup, everything seems good now. Appreciate you getting to this so fast!

**Comment by vecima** [comment]
Hi, since about 2-3 months ago no users can log into my server using Steam. I thought it may be related to this issue, but I've tried 1.0.2 and 2.0.1, and I'm constantly running into a problem where a (generated) decode method winds up throwing an index out of range exception. parseEncryptedAppTicket then catches the exception and returns null.

I ran into this same problem ~3 years ago, and don't recall how it was eventually resolved.

Should I not be setting the size to 1024 when I request the ticket? in testing the token does seem to have a lot of "00" towards the end, so maybe a different size would be better? On the steam doc example they use 1024. I can see why the decode method hits this error - it moves the buffer position twice in the loop so eventually it tries to skip when it's already at the end.

`'index out of range: 1024 + 1 > 1024'`

here's the stack at that point:
'RangeError: index out of range: 1024 + 1 > 1024\n
    at indexOutOfRange (D:\\projects\\type3-services\\node_modules\\protobufjs\\src\\reader.js:13:12)\n
    at BufferReader.skip (D:\\projects\\type3-services\\node_modules\\protobufjs\\src\\reader.js:349:23)\n
    at BufferReader.Reader.skipType (D:\\projects\\type3-services\\node_modules\\protobufjs\\src\\reader.js:363:18)\n
    at Type.EncryptedAppTicket$decode [as decode] (eval at Codegen (D:\\projects\\type3-services\\node_modules\\@protobufjs\\codegen\\index.js:50:33), <…1\n
    at new Promise (<anonymous>)\n
    at __awaiter (D:\\projects\\type3-services\\node_modules\\@feathersjs\\authentication\\lib\\core.js:4:12)\n
    at Object.authenticate (D:\\projects\\type3-services\\node_modules\\@feathersjs\\authentication\\lib\\core.js:143:16)\n
    at Object.<anonymous> (D:\\projects\\type3-services\\node_modules\\@feathersjs\\authentication\\lib\\service.js:81:43)\n
    at Generator.next (<anonymous>)\n    at D:\\projects\\type3-services\\node_modules\\@feathersjs\\authentication\\lib\\service.js:8:71'

**Comment by DoctorMcKay** [comment]
@vecima You need to pass only the bytes that contain the ticket. 1024 bytes should be plenty big enough to contain the ticket, but you need to check the actual length of the ticket.

The actual length of the ticket is available from the `pcbTicket` argument to [GetEncryptedAppTicket](https://partner.steamgames.com/doc/api/ISteamUser#GetEncryptedAppTicket).

**Comment by vecima** [comment]
@DoctorMcKay Thanks for the information. I used the pcbTicket arg to shorten the token to the appropriate length, and after that I'm able to parse the steam tickets when running my server locally.

For some reason it's not working on my server in production. I verified in package.json that steam-appticket has "version": "^2.0.1" and in package-lock.json that it has "version": "2.0.1". I also deleted node_modules and ran "npm install", but when I try to authenticate I'm still getting null returned for the ticket. but again, only on my production Ubuntu server.

EDIT:

I figured it out, I was running the app as a systemd service on my prod server and forgot that the service file was specifying an older version of node, that must have had the older version of the steam-appticket library running with it. After updating the .service file to point to the updated node everything is working properly.
